core/src/commonMain/kotlin/com/ustadmobile/core/controller/PersonEditPresenter.kt

```diff
@@ -2,38 +2,53 @@ package com.ustadmobile.core.controller
 
 import com.ustadmobile.core.db.UmAppDatabase
 import com.ustadmobile.core.generated.locale.MessageID
+import com.ustadmobile.core.impl.AppConfig
+import com.ustadmobile.core.impl.UstadMobileSystemImpl
 import com.ustadmobile.core.util.DefaultOneToManyJoinEditHelper
 import com.ustadmobile.core.util.MessageIdOption
 import com.ustadmobile.core.util.ext.enrolPersonIntoClazzAtLocalTimezone
 import com.ustadmobile.core.util.ext.putEntityAsJson
 import com.ustadmobile.core.util.ext.setAttachmentDataFromUri
+import com.ustadmobile.core.view.ContentEntryListTabsView
 import com.ustadmobile.core.view.PersonEditView
 import com.ustadmobile.core.view.UstadEditView.Companion.ARG_ENTITY_JSON
+import com.ustadmobile.core.view.UstadView
 import com.ustadmobile.core.view.UstadView.Companion.ARG_ENTITY_UID
 import com.ustadmobile.door.DoorLifecycleOwner
 import com.ustadmobile.door.doorMainDispatcher
 import com.ustadmobile.lib.db.entities.ClazzMemberWithClazz
 import com.ustadmobile.lib.db.entities.Person
 import com.ustadmobile.lib.db.entities.PersonPicture
+import com.ustadmobile.lib.db.entities.WorkSpace
 import com.ustadmobile.lib.util.getSystemTimeInMillis
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
 import kotlinx.coroutines.withTimeoutOrNull
 import kotlinx.serialization.builtins.list
 import kotlinx.serialization.json.Json
 import org.kodein.di.DI
+import org.kodein.di.instance
 
 
 class PersonEditPresenter(context: Any,
                           arguments: Map<String, String>, view: PersonEditView, di: DI,
                           lifecycleOwner: DoorLifecycleOwner)
     : UstadEditPresenter<PersonEditView, Person>(context, arguments, view, di, lifecycleOwner) {
 
+    private lateinit var serverUrl: String
+
+    private val impl: UstadMobileSystemImpl by instance()
+
+    private  lateinit var nextDestination: String
+
     override val persistenceMode: PersistenceMode
         get() = PersistenceMode.DB
 
-    val clazzMemberJoinEditHelper = DefaultOneToManyJoinEditHelper<ClazzMemberWithClazz>(ClazzMemberWithClazz::clazzMemberUid,
+    private var workSpace: WorkSpace = WorkSpace().apply {
```

Comment:
Why is the workspace object needed here? Couldn't we just have a simple boolean registrationMode as true/false ? It seems like we are using the Workspace class when all we need from this is the boolean.